src/kbd.rs

```diff
@@ -0,0 +1,274 @@
+use x86::io::inb;
+
+// PC keyboard interface constants
+
+const KBSTATP: u16    = 0x64;    // kbd controller status port(I)
+const KBS_DIB: u8    = 0x01;    // kbd data in buffer
+const KBDATAP: u16    = 0x60;    // kbd data port(I)
+
+const NO: u8         = 0;
+
+const SHIFT: u8      = (1<<0);
+const CTL: u8        = (1<<1);
+const ALT: u8        = (1<<2);
+
+const CAPSLOCK: u8   = (1<<3);
+const NUMLOCK: u8    = (1<<4);
+const SCROLLLOCK: u8 = (1<<5);
+
+const E0ESC: usize      = (1<<6);
+
+// Special keycodes
+const KEY_HOME: u8   = 0xE0;
+const KEY_END: u8    = 0xE1;
+const KEY_UP: u8     = 0xE2;
+const KEY_DN: u8     = 0xE3;
+const KEY_LF: u8     = 0xE4;
+const KEY_RT: u8     = 0xE5;
+const KEY_PGUP: u8   = 0xE6;
+const KEY_PGDN: u8   = 0xE7;
+const KEY_INS: u8    = 0xE8;
+const KEY_DEL: u8    = 0xE9;
+
+#[deny(const_err)]
+macro_rules! C {
+    ($c:expr) => {
+        $c as u8 - '@' as u8
+    }
+}
+
+static NORMALMAP: [u8; 256] = [
+  NO,   0x1B, '1' as u8,  '2' as u8,  '3' as u8,  '4' as u8,  '5' as u8,  '6' as u8,  // 0x00
```

Comment:
I think if instead of `char as u8` we can do `b'char'`. Just off the top of my head, but might be worth trying so the `as u8` doesn't dominate so much of the page (and less typing for future device drivers).

For example, `'1' as u8` could be `b'1'`